### PR TITLE
#160706464-API endpoint for user Login

### PR DESCRIPTION
**Pull request by fejiroofficial**
**What does this PR do?**
This is a pull request for the API login endpoint for the application. 

**Description of Task to be completed?**
Have the following endpoint working and tested on postman:
`POST /api/v1/auth/login`

**How should this be manually tested?**
- Clone the repo using git clone` https://github.com/fejiroofficial/Fast-Food-Fast.git.`
- run `git checkout ft-login-160706464`
- run `npm install` and `npm start`.
- `npm run migrate`
- `to run the test: npm run test`
- `using postman`

screenshot 194

**Relevant pivotal stories**
[160706464](https://www.pivotaltracker.com/n/projects/2196470/stories/160706464)

### CODE REVIEW

File: server/controllers/users.js [hunk 1]
```diff
@@ -63,6 +65,50 @@ class UserController {
         });
       }));
   }
+
+  /**
+* @function login
+* @memberof UserController
```
Comment (EricDav) [inline server/controllers/users.js:71]: No params

File: server/controllers/users.js [hunk 2]
```diff
@@ -63,6 +65,50 @@ class UserController {
...
+          return res.status(401).json({
+            success: 'false',
+            message: 'no account created with this email address.',
+          });
+        }
+        const allowEntry = bcrypt.compareSync(password, user.password);
+        if (!allowEntry) {
+          return res.status(401).json({
+            success: 'false',
+            message: 'Incorrect password!',
```
Comment (EricDav) [inline server/controllers/users.js]: The message is too specific, it might give hint to hackers

File: server/middlewares/validateLogin.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+const signinError = (message) => {
```
Comment (EricDav) [inline server/middlewares/validateLogin.js:11]: No Jsdocs

**Comment by EricDav** [comment]
`What does this PR do?` is not saying what it does

File: server/controllers/users.js (hunk 1 above)
Comment (fejiroofficial) [reply server/controllers/users.js:71]: done

File: server/controllers/users.js (hunk 2 above)
Comment (fejiroofficial) [reply server/controllers/users.js]: You have entered an invalid email or password

File: server/middlewares/validateLogin.js (hunk 3 above)
Comment (fejiroofficial) [reply server/middlewares/validateLogin.js:11]: done

**Review by EricDav — APPROVED** [review]

**Pull request merged**